tools/utils.py

```diff
@@ -196,3 +198,46 @@ def dict_to_ascii(input):
 def json_file_to_dict(fname):
     with open(fname, "rt") as f:
         return dict_to_ascii(json.load(f, object_pairs_hook=OrderedDict))
+
+# Wowza, double closure
+def argparse_type(casedness, prefer_hyphen=False) :
+    def middle(list, type_name):
+        def parse_type(string):
+            if prefer_hyphen: newstring = casedness(string).replace("_","-")
+            else:             newstring = casedness(string).replace("-","_")
+            if newstring not in list:
+                raise argparse.ArgumentTypeError("{0} is not a supported {1}. Supported {1}s are:\n{2}".format(string, type_name, columnate(list)))
+            else:
+                return newstring
+        return parse_type
+    return middle
+
+argparse_uppercase_type = argparse_type(str.upper, False)
+argparse_lowercase_type = argparse_type(str.lower, False)
+argparse_uppercase_hyphen_type = argparse_type(str.upper, True)
+argparse_lowercase_hyphen_type = argparse_type(str.lower, True)
+
+def argparse_many(fn):
+    def wrap(string):
+        return [fn(s) for s in string.split(",")]
+    return wrap
+
+def argparse_filestring_type(string) :
+    if exists(string) : return string
```

Comment:
@theotherjimmy For coding style please follow

```
if a:
   b
else:
   c
```

Not big fan of it either but it's consistent with the rest of the codebase